services/crypto_approve_allowance.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package proto;
+
+/*-
+ * ‌
+ * Hedera Network Services Protobuf
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+option java_package = "com.hederahashgraph.api.proto.java";
+option java_multiple_files = true;
+
+import "basic_types.proto";
+
+/**
+ * Create a series of hbar/token approved allowances. Each allowance grants a spender the right
+ * to transfer a pre-determined amount of the owner's hbar/token to any other account of their
+ * choice.
+ */
+message CryptoApproveAllowanceTransactionBody {
+    message CryptoApproval {
+        /**
+         * The account ID of the spender of the hbar allowance.
+         */
+        AccountID spender = 1;
+
+        /**
+         * The amount of the spender's allowance in tinybars.
+         */
+        int64 amount = 2;
+    }
+
+    message TokenApproval {
+        /**
+         * The token that the allowance pertains to.
+         */
+        TokenID tokenId = 1;
+
+        /**
+         * The account ID of the token allowance spender.
+         */
+        AccountID spender = 2;
+
+        oneof allowance {
+            /**
+             * The amount of the spender's token allowance.
+             */
+            int64 amount = 3;
+
+            /**
+             * If true, the spender has access to all of the account owner's NFT instances (currently
+             * owned and any in the future).
+             */
+            bool approvedForAll = 4;
+        }
+    }
+
+    /**
+     * List of hbar allowances approved by the account owner.
```

Comment:
Should we disclose here that there's some limitations on the total number of allowances allowed in a single transaction?